--- a/scripts/copycat_generate_results.sh
+++ b/scripts/copycat_generate_results.sh
@@ -8,8 +8,8 @@
 
 capture_pane() {
 	local file=$1
-	# copying 9M lines back will hopefully fetch the whole scrollback
-	tmux capture-pane -S -9000000 -p > "$file"
+	local height="$(tmux display-message -p '#{pane_height}')"
+	tmux capture-pane -S "-$height" -p > "$file"
 }
 
 # doing 2 things in 1 step so that we don't write to disk too much
